shell/platform/darwin/graphics/FlutterDarwinContextMetal.mm

```diff
@@ -33,32 +33,39 @@ - (instancetype)initWithMTLDevice:(id<MTLDevice>)device
 
     if (!_device) {
       FML_DLOG(ERROR) << "Could not acquire Metal device.";
-      [self release];
       return nil;
     }
 
     _commandQueue = commandQueue;
 
     if (!_commandQueue) {
       FML_DLOG(ERROR) << "Could not create Metal command queue.";
-      [self release];
       return nil;
     }
 
     [_commandQueue setLabel:@"Flutter Main Queue"];
 
+    CVReturn cvReturn = CVMetalTextureCacheCreate(kCFAllocatorDefault,  // allocator
+                                                  nil,      // cache attributes (nil default)
+                                                  _device,  // metal device
+                                                  nil,      // texture attributes (nil default)
+                                                  &_textureCache  // [out] cache
```

Comment:
CoreFoundation objects need to be explicitly released. IIRC, this would leak the texture cache in the dealloc unless `__bridge_transfer`ed

Reply:
I was worried about the same. I will try to add a test to see that everything that was allocated gets collected.

Reply:
I wasn't able to come up with a way to test de-alloc in a nice way. I audited the code and made sure that are CoreFoundation objects are released and retained as necessary.